src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java

```diff
@@ -335,10 +338,32 @@ public ResponseEntity<GenerateAssetResponseDTO> generateAsset(
             throw new NotFoundException("Source Container: " + generateAssetDTO.getSourceContainer());
         }
 
+        // 404s of there are no mp4 files in the container
+        // this is called in the MK process anyway but AMS doesn't need the specific filename
+        azureFinalStorageService.getMp4FileName(generateAssetDTO.getSourceContainer());
+
         log.info("Attempting to generate asset: {}", generateAssetDTO);
 
         var result = mediaServiceBroker.getEnabledMediaService().importAsset(generateAssetDTO);
         if (result.getJobStatus().equals(JobState.FINISHED.toString())) {
+            // add new version to recording etc
+            var parentRecording = recordingService.findById(UUID.fromString(generateAssetDTO.getSourceContainer()));
```

Comment:
Looking at the edit tool documentation, it looks like source container will not be the id of the recording- that is only part of the name:  i.e. the doc gives this as an example for the source container: `"sourceContainer": "1c92c5ea-3c7f-44d1-8052-fa8989a5c28c-input"`. 

Might want a way to get the id from the container name to avoid an error. 

The container with the plain id as the name would be the original recording so cannot just change the naming convention to only the id.

Reply:
thanks, that wasn't clear on the popup box in the app. I'll update it